src/core/Microsoft.PowerToys.Settings.UI.Lib/GeneralSettings.cs

```diff
@@ -98,7 +98,7 @@ public bool UpgradeSettingsConfiguration()
         {
             try
             {
-                if (Helper.CompareVersions(PowertoysVersion, Helper.GetProductVersion()) < 0)
+                if (Helper.CompareVersions(PowertoysVersion, Helper.GetProductVersion()) != 0)
```

Comment:
It's nice to have this so that the settings file is of the correct version if someone chooses to downgrade. However, do we want to add tests so that everything works as expected when we downgrade?

Reply:
I don't think that is possible with a unit test.   This change wasn't to support downgrading becuase the code being tested will always be the latest.  However if we have experimental releases in the wild this might be necessary as we This change was necessary because at the time of writing we released an experimental .22 release, but the current product version was .21.  This would force the tests that validated that we serialized the current version to fail.   We could either remove the .22 data from this test as it's experimental or leave it as is.     I personally thinks it makes sense as is because we want to support the .22 -> .23  path as well.  Plus once we have dev product versions in the pipeline we will still probably want this logic.